### PR TITLE
[F] Add created by me filter to projects lists

### LINKED ISSUE #1954: Add mechanism to view "My Projects" to the back end
## Feature Description

A quick way to view and access all the projects I have created in the back end.

## *Why* is this feature important? *Who* does it help?

This feature would particularly help users who are uploading many texts, will be returning to a project to continue to build it over a longer period of time, or will often want to return to a text that they uploaded a while ago to make changes. 

## User Stories

I am an instructor who is putting all of the novels I am teaching this semester on Manifold. I will be working on this over the course of several months, so the texts that I previously ingested will no longer be in the "Recently Updated" list, but I'd like a quick way to access them.

### PR DESCRIPTION
**Pull request by SMaxOwok**
Closes #1954

### CODE REVIEW

**Comment by SMaxOwok** [comment]
Let me know what you think about this.  

Alternatives could be:
- Not changing `Filterable` and calling the scope `with_creating_ability`.
- Calling the scope `by_creator` and passing an ID as the value.  FWIW I think this is not a good approach, especially given the recent change to how filters are made in the client.

**Comment by zdavis** [comment]
Let's name the scope `with_creator_role` and adjust the method that checks the scope type to:

```
    def scope_requires_user?(key)
      key.start_with?("with_") && (key.end_with?("_role") || key.end_with?("_ability")) 
    end
```
I might also change the method name a bit to be more specific. It's a little hard to understand at a glance what `user_scope?` means.

File: api/app/models/project.rb [hunk 1]
```diff
@@ -170,6 +170,8 @@ class Project < ApplicationRecord
     order(by)
   }
 
+  scope :with_creator_role, ->(user = nil) { puts "ASDFASDFASdf"; where(creator: user) if user.present? }
```
[the remark this answers is not in the document]
Comment (SMaxOwok) [reply api/app/models/project.rb]: ooooops

**Comment by zdavis** [comment]
Probably just need to rebase on master to fix the failing snapshots?

**Comment by SMaxOwok** [comment]
Indeed

**Pull request merged**